tkg/client/audit_upgrade_helper.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package client
+
+import (
+	"encoding/base64"
+	"strings"
+
+	"github.com/pkg/errors"
+	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
+)
+
+const (
+	auditConfigFilePath = "/etc/kubernetes/audit-policy.yaml"
+	apiVersionV1Alpha1  = "apiVersion: audit.k8s.io/v1alpha1"
```

Comment:
Do we have v1alpha1 version of auditconfig? v1 might be promoted from v1beta1 so their spec are identical. I am not sure if v1alpha1 's spec is the same as v1's.

Reply:
in legacy ytt provider folder, we don't have v1alpha1, only have v1beta1; but from upstream codes, both v1alpha1 and v1beta1 are replaced by v1  in 1.24.x, so if users customized to use v1alpha1, we will replace it with v1 here.

v1apha1 is almost same as v1beta1, but there are some comments change.
v1 has added some fiels compared to v1beta1

looks good based on above information? feel free to comment if there are concerns, thanks.

Reply:
Thanks. Looks good.